packages/react-query/src/useBaseQuery.ts

```diff
@@ -48,6 +48,12 @@ export function useBaseQuery<
         'Bad argument type. Starting with v5, only the "Object" form is allowed when calling query related functions. Please use the error stack to find the culprit call. More info here: https://tanstack.com/query/latest/docs/react/guides/migrating-to-v5#supports-a-single-signature-one-object',
       )
     }
+    const defaultQueryFn = queryClient?.getDefaultOptions().queries?.queryFn
```

Comment:
this isn’t enough because defaults can be defined on different “levels”.

the correct way is to check if there is a `queryFn` in `defaultedOptions` after line 62, where we do:

```
const defaultedOptions = client.defaultQueryOptions(options)
```

Reply:
Thanks for pointing that out! I’ve moved the check after line 62 as you suggested.
Do you think an environment check `(process.env.NODE_ENV !== 'production')` is necessary here?

Also, would it be okay not to throw an error in this case?
If someone provides `initialData` and sets a long `staleTime`, it could potentially cause issues.

Reply:
please do it like everywhere else: with an env check + a console error.

Reply:
thank you. I’ve updated the code.